app/views/spree/users/_past_orders.html.haml

```diff
@@ -8,16 +8,16 @@
         %th.order4.show-for-large-up= t('.items')
         %th.order5.text-right= t('.total')
         %th.order6.text-right.show-for-large-up= t('.paid?')
-        %th.order7.text-right= t('.view')
+        %th.order7.text-right= t('.status')
       %tbody.transaction-group{"ng-repeat" => "order in Orders.all | filter:{changes_allowed:false} as pastOrders", "ng-class-odd"=>"'odd'", "ng-class-even"=>"'even'"}
         %tr.order-row
-          %td.order1
-            %a{"ng-href" => "{{::order.path}}", "ng-bind" => "::order.number"}
+          %td.order1{"ng-bind" => "::order.number"}
           %td.order2
             %a{"ng-href" => "{{::Orders.shopsByID[order.shop_id].hash}}#{main_app.shop_path}", "ng-bind" => "::Orders.shopsByID[order.shop_id].name"}
           %td.order3.show-for-large-up{"ng-bind" => "::order.completed_at"}
           %td.order4.show-for-large-up{"ng-bind" => "::order.item_count"}
           %td.order5.text-right{"ng-class" => "{'debit': order.payment_state != 'paid', 'credit': order.payment_state == 'paid'}","ng-bind" => "::order.total | localizeCurrency"}
           %td.order6.text-right.show-for-large-up{"ng-class" => "{'debit': order.payment_state != 'paid', 'credit': order.payment_state == 'paid'}", "ng-bind" => "::(order.payment_state == 'paid' ? 'say_yes' : 'say_no') | t"}
-          %td.order7.text-right
-            %a{"ng-href" => "{{::order.path}}" }= t('.view')
+            %td.order7.text-right{ "ng-switch" => "::order.state"  }
+              %a{ "ng-switch-when" => "complete", "ng-href" => "{{::order.path}}" }Completed
+              %span{ "ng-switch-when" => "canceled" }Cancelled
```

Comment:
Completed and Cancelled should also be translatable.
```suggestion
              %a{ "ng-switch-when" => "complete", "ng-href" => "{{::order.path}}" }= t(".completed")
              %span{ "ng-switch-when" => "canceled" }= t(".cancelled")
```
Plus the corresponding additions to en.yml.

Reply:
Hey I made the requested changes but forgot to reply here! Could you check and merge this if everything's inorder or let me know if something else is required.